crypto/libsodium-fork/src/libsodium/crypto_core/ed25519/ref10/ed25519_ref10_sc.c

```diff
@@ -0,0 +1,299 @@
+#include "private/ed25519_ref10_sc.h"
+
+#include <string.h>
+
+
+typedef unsigned __int128 uint128_t;
+#define HAVE_UINT128
```

Comment:
Diffed this file against libdonna modm-donna-64bit.h, all looks the same. This HAVE_UINT128 define is not needed, however, you brought over all the code that would have checked it without the ifdefs.